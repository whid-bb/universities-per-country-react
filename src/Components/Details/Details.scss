.details-header {
  width: 100%;
  text-align: center;
  padding-block: 1rem;
  font-weight: 800;
  border-bottom: 1px solid #ccc;

  span {
    margin-right: 10px;
  }
}
