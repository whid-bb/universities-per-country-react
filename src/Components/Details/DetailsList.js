import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';
import LinkIcon from '@mui/icons-material/Link';
import SchoolIcon from '@mui/icons-material/School';
import './DetailsList.scss';

function DetailsList({ name, webPages, animDelay }) {
  const [show, setShow] = useState('');

  useEffect(() => {
    setShow('showList');
  }, []);

  return (
    <ul className={`details-list ${show}`}>
      <li
        className="details-item"
        style={{ animationDelay: `${animDelay}ms` }}
      >
        <p className="u-name">
          <SchoolIcon />
          {name}
        </p>
        <p className="u-link">
          <LinkIcon />
          <a target="_blank" rel="noreferrer" href={webPages}>{webPages}</a>
        </p>
      </li>
    </ul>
  );
}

DetailsList.propTypes = {
  name: PropTypes.string.isRequired,
  webPages: PropTypes.string.isRequired,
  animDelay: PropTypes.number.isRequired,
};

export default DetailsList;
