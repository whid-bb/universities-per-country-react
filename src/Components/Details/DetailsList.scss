.details-list {
  opacity: 0;
  transition: all 300ms ease-in;

  &.showList {
    opacity: 1;
  }
}

.details-item {
  animation-name: fadeIn;
  animation-duration: 300ms;
  animation-fill-mode: forwards;
  animation-timing-function: cubic-bezier(0.075, 0.82, 0.165, 1);
  opacity: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border-bottom: 1px solid #ccc;
  padding: 1rem;

  > p {
    display: flex;
    align-items: center;
    margin-bottom: 10px;

    &:last-child {
      margin-bottom: 0;
    }

    svg {
      margin-right: 10px;
    }
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
    transform: translateY(-20px);
  }

  to {
    opacity: 1;
    transform: translateX(0);
  }
}
