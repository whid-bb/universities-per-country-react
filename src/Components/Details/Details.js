/* eslint-disable camelcase */

import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchCountriesDetails } from '../../redux/countries/CountriesService';
import { searchCountries } from '../../redux/countries/Countries';
import DetailsList from './DetailsList';
import './Details.scss';

function Details() {
  const { details } = useSelector((state) => state.countries);
  const { country } = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(searchCountries({ name: country }));
    dispatch(fetchCountriesDetails(country));
  }, []);

  return (
    <>
      <div className="details-header">
        <span>
          Universities:
        </span>
        {details.length}
      </div>
      {details.map(({ name, web_pages }, i) => (
        <DetailsList key={name} animDelay={i * 100} name={name} webPages={web_pages[0]} />
      ))}
    </>
  );
}

export default Details;
