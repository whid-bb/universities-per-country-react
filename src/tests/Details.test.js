import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Details from '../Components/Details/Details';

describe('Details test', () => {
  it('Details render correctly', () => {
    const DetailsPage = renderer.create(
      <Provider store={store}>
        <Details />
      </Provider>,
    ).toJSON();
    expect(DetailsPage).toMatchSnapshot();
  });

  test('Render Details component', () => {
    render(
      <Provider store={store}>
        <Details />
      </Provider>,
    );
  });
});
