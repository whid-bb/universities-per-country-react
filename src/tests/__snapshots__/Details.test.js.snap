// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test Details render correctly 1`] = `
<div
  className="details-header"
>
  <span>
    Universities:
  </span>
  0
</div>
`;
